pointcloud/src/main/scala/geotrellis/pointcloud/raster/ept/DEMRasterSource.scala

```diff
@@ -102,16 +102,124 @@ case class DEMRasterSource(
         pipeline.execute
 
         val pointViews = pipeline.getPointViews().asScala.toList
-        val viewSizes = pointViews.map(_.length)
-        val (nCoords, starts) = viewSizes.foldLeft(0 -> List.empty[Int]) { case ((acc, cum), v) => (v + acc, cum :+ acc) }
-        val coords = Array.ofDim[Coordinate](nCoords)
-        viewSizes.zip(starts).zip(pointViews).foreach { case ((n, z), pv) =>
-          val pc = pv.getPointCloud(DimType.X, DimType.Y, DimType.Z)
-          cfor(0)(_ < n, _ + 1) { i => coords(z + i) = pc.getCoordinate(i) }
+        assert(pointViews.length == 1, "Triangulation pipeline should have single resulting point view")
```

Comment:
Oops, I think we can not do that in the semantics of the current read function. 

The type signature of this read function implies that `None` is an expected behavior and we don't expect it to throw. Mb just adding a `logger.warn` instead of throwing here would be enough, otherwise `Option[Raster]` signature is completely broken.

For the context: in the future RasterSources would be abstracted over the effect type: [GeoTiffRasterSource example](https://github.com/geotrellis/geotrellis-contrib/blob/develop/vlm/src/main/scala/geotrellis/contrib/vlm/effect/geotiff/GeoTiffRasterSource.scala#L74) and a [RasterSource interface example](https://github.com/geotrellis/geotrellis-contrib/blob/develop/vlm/src/main/scala/geotrellis/contrib/vlm/effect/RasterSourceF.scala), so users would have control over the effect (Either / Option / IO / Task / etc).

Reply:
In this case, I'd rather not lose the explicit check.  This is a weird edge case that we don't know much about.  Rather than a whisper, I want this code to scream when we find a test case of this.